torch/csrc/jit/script/compiler.cpp

```diff
@@ -1597,10 +1597,41 @@ struct to_ir {
   // else
   //   raise Exception("Hi")
   // print(a)
-  void emitRaise(const SourceRange& loc) {
+  void emitRaise(const Raise& stmt) {
+    if (stmt.expr().present()) {
+      auto exceptionNode = stmt.expr().get().tree();
+      // Verifies that the expression is of the form `raise f(expr, ...)`, and
+      // outputs "<f>: <expr>"
+      if (exceptionNode->kind() == TK_APPLY &&
+          !Apply(exceptionNode).inputs().empty()) {
+        auto apply = Apply(exceptionNode);
+        if (apply.callee().kind() == TK_VAR) {
+          auto exception_name = Var(apply.callee()).name().name();
+          auto exception_arg_expr = Expr(apply.inputs()[0]);
+          auto exception_msg = asSimple(emitSugaredExpr(exception_arg_expr, 1));
+          if (!exception_msg->type()->isSubtypeOf(StringType::get())) {
+            exception_msg =
+                graph->insert(prim::str, {exception_msg}, {}, stmt.range());
+          }
+          auto separator = insertConstant(*graph, ": ", nullptr, stmt.range());
```

Comment:
It would be better to do this string formatting in RaiseException and have the generic version just pass generic values for it. Eventually we would like to re-raise the right python exception so having it separate would help here.

Reply:
Oh that's 10x better.